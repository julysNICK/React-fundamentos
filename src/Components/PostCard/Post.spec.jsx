import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';
const props = postCardPropsMock;
describe('<PostCard />', () => {
  it('should render Postcard correctly', () => {
    render(<PostCard {...props} />);

    expect(screen.getByAltText(/title 1/i)).toHaveAttribute('src', 'img/img.png');

    expect(screen.getByRole('heading', { name: 'title 1 1' })).toBeInTheDocument();

    expect(screen.getByText('body 1')).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
