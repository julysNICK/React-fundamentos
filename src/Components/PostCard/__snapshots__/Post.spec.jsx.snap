// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
       
      title 1
       
      1
    </h1>
      
    <p>
       
      body 1
       
    </p>
  </div>
</div>
`;
